### PR TITLE
Add Syntax Check to Slovakia

### PR DESCRIPTION
**Pull request by iekadou**
Add Slovakia TIN Check Digit

### CODE REVIEW

File: src/CountryHandler/Slovakia.php [hunk 1]
```diff
@@ -24,14 +24,28 @@ final class Slovakia extends CountryHandler
...
+
+    public function hasValidRule(string $tin): bool
+    {
+        if (strlen($tin) === 10
+            && (int) $tin % 11 === 0) {
+            return true;
+        }
+        if (strlen($tin) === 10) {
+            return ((int) substr($tin, 0, 9) % 11) % 10 === (int) substr($tin, 9, 1);
+        }
```
Comment (drupol) [inline src/CountryHandler/Slovakia.php:48]:
```suggestion
        if (strlen($tin) === 10) {
          if ( (int) $tin % 11 === 0) {
            return true;
          }
          
          return ((int) substr($tin, 0, 9) % 11) % 10 === (int) substr($tin, 9, 1);        
        }
```

**Review by drupol — CHANGES_REQUESTED** [review]
Thanks.

Do you have documentation for this?

File: src/CountryHandler/Slovakia.php (hunk 1 above)
Comment (iekadou) [reply src/CountryHandler/Slovakia.php:48]: On the EU checker page (https://ec.europa.eu/taxation_customs/tin/#/check-tin) you can try e.g. 2822030541
This TIN is valid, even if it has 10 digits and starts with < 54.

We decompiled the Java lib you can find here:
https://taxation-customs.ec.europa.eu/document/download/07664928-b4f3-47a5-81f7-438767dc01cb_en?v=1721203258495

The modulo 11 check has those 2 modes in the decompiled lib.

File: src/CountryHandler/Slovakia.php [hunk 2]
```diff
@@ -24,12 +26,28 @@ final class Slovakia extends CountryHandler
...
+        if (strlen($tin) === 10
+            && (int) $tin % 11 === 0) {
+            return true;
+        }
+
+        if (strlen($tin) === 10) {
+            return (int) substr($tin, 9, 1) === ((int) substr($tin, 0, 9) % 11) % 10;
+        }
+
+        return strlen($tin) === 9;
```
Comment (drupol) [inline src/CountryHandler/Slovakia.php:42]:
```suggestion
        if (strlen($tin) === 10) {
          if ((int) $tin % 11 === 0) {
            return true;
          }

          return (int) substr($tin, 9, 1) === ((int) substr($tin, 0, 9) % 11) % 10;
        }

        return strlen($tin) === 9;
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: src/CountryHandler/Slovakia.php (hunk 2 above)
Comment (drupol) [reply src/CountryHandler/Slovakia.php:42]: (don't forget to run `phpcsfixer` after that change)

Comment (iekadou) [reply src/CountryHandler/Slovakia.php:42]: Adapted your solution in my latest commit.

Comment (drupol) [reply src/CountryHandler/Slovakia.php:42]: I don't think so, or I misunderstand what you're doing.

There are still 2 calls to `if (strlen($tin) === 10)` and we definitely should avoid repeating ourselves.

Comment (iekadou) [reply src/CountryHandler/Slovakia.php]: Commit was not pushed 👍

Comment (iekadou) [reply src/CountryHandler/Slovakia.php]: I also removed the strlen check as we did that before in the matchLength.

Comment (drupol) [reply src/CountryHandler/Slovakia.php]: That was my next concern :)

Comment (iekadou) [reply src/CountryHandler/Slovakia.php]: Should be pushed now :)

**Comment by drupol** [comment]
Thank you!

**Comment by iekadou** [comment]
@drupol Can we do a new release for this? :)

**Comment by drupol** [comment]
Yes that's the plan. I have a few quirks to fix before, I will try to take care of this after my day, in the evening.

**Pull request merged**